src/imgui/ImGuiWaveViewer.cc

```diff
@@ -123,14 +123,28 @@ static void plotHistogram(std::span<const float> values, float scale_min, float
 }
 
 
+#define SHOW_CHIP_KEY "showChip"
+
 void ImGuiWaveViewer::save(ImGuiTextBuffer& buf)
 {
 	savePersistent(buf, *this, persistentElements);
+	for (const auto& chip: openChips) {
+		buf.appendf(SHOW_CHIP_KEY"=%s\n", chip.c_str());
+	}
+}
+
+void ImGuiWaveViewer::loadStart()
+{
+	openChips.clear();
 }
 
 void ImGuiWaveViewer::loadLine(std::string_view name, zstring_view value)
 {
-	loadOnePersistent(name, value, *this, persistentElements);
+	if (loadOnePersistent(name, value, *this, persistentElements)) {
+		// already handled
+	} else if (name == SHOW_CHIP_KEY) {
+		openChips.insert(std::string(value));
```

Comment:
Detail. Instead of converting `zstring_view` into `string` and then placing a copy of that string into the `set` (maybe later a `map`?), you could directly construct the string in-place via: `openChips.emplace(value)`  (caveat: I have not tested it).   (or later when you use a `map` via `openChips.try_emplace(value, true)`).

Reply:
Good point, changed it.